### ISSUE TITLE
Allow for specification of  encoding in getCodes() (RJSDMX)

### ISSUE THREAD

**Issue by Tungurahua** [amattioc/SDMX#40]
When I request Codes from Eurostat in `R` using `RJSDMX`, non-Ascii characters are not displayed correctly. See the following example:
```
> c_GEO <- getCodes("EUROSTAT","demo_r_pjanaggr3","GEO")
> c_GEO[[2]] # should return "Thüringen"
[1] "Th√ºringen"
```
I ran the result through all available encodings, but didn't succeed to retrieve the correct value
```
> str <- sapply(iconvlist(),function(x){iconv(c_GEO[[2]], from=x, to="")[[1]]})
> grep("Thüringen",str,value = T)
named character(0)
```
Would it be possible to pass an encoding as argument to `getCodes()` and similar functions to remedy this situation?

I am aware that this might be a problem of my specific setup, however I am unsure which parameters to report. Please let me know if you need additional session-info. 

Apart from that: Great package! Thanks for the effort!

**Comment by amattioc** [comment]
Hi, thanks for the feedback. You're right, I see that multibyte strings work fine in a full UTF8 environment, but they can contain junk characters in other configurations. 

While I investigate, can you send me the output of

` Sys.getlocale()`

**Comment by amattioc** [comment]
I found a bug in the handling of streams from providers that support compression. I think this solves your problem too. I uploaded the fix in commit f7d343e73a2018ed2119a0fec9385aaa47930d4c

Can you check that, with this update, multibyte strings now show correctly in your env?

Thanks, Attilio

**Comment by Tungurahua** [comment]
Hi Attilio, all works fine now!

Thanks a lot,
Albrecht
